cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/jms/JndiBasedJmsProvider.java

```diff
@@ -44,7 +44,8 @@
   private final ConnectionFactory connectionFactory;
   private final Destination destination;
 
-  public JndiBasedJmsProvider(Hashtable<String, String> initialContextEnv, String destinationName) {
+  public JndiBasedJmsProvider(Hashtable<String, String> initialContextEnv, final String destinationName,
```

Comment:
Unfortunately the stupid javax InitialContext only takes HashTable =(

So I have to use HashTable all the way.